### DISCUSSION TITLE
Neither building an application nor the binary with GCC works on Windows

### DISCUSSION THREAD

**Discussion by KaySteinhoff** [webui-dev/webui#475] [Q&A]
I've used WebUI on my Linux machine and wanted to build on Windows, just to test everything, using this command:

```
gcc src/*.c -I include -L lib -lwebui-2 -Wall -Werror -o app
```

This works flawlessly on Linux but throws this error on Windows:
```
lib/webui-2.dll: file not recognized: File format not recognized
collect2.exe: error: ld returned 1 exit status
```

Now no issue I'll just rebuild from source I thought but both make and mingw32-make threw errors. Now I'm kinda stuck as I'm not really familiar with make and confused as I tried to use both the 2.5.0 beta and 2.4.2 release versions and both threw the compiler/make error(s).

Is this an issue or just me not using the binary correctly?

**Comment by AlbertShown** [comment]
To build the static version of your project in Windows, using MinGW:

`gcc -Os -Wl,-subsystem=windows main.c -I"include/" -L"lib" -lwebui-2-static -lws2_32 -Wall -luser32 -static -o main.exe`

To build the dynamic version of your project in Windows, using MinGW:

`gcc -Wl,-subsystem=windows main.c -I"include/" -L"lib" "webui-2.dll" -lws2_32 -Wall -luser32 -o main-dyn.exe`

**Reply by KaySteinhoff** [reply]
Using the static version I get undefined references to the WebUI function calls and the dynamic version just results in the same file format error as in my original post.

Here's the entire gcc command:
```
gcc -Os -Wl,-subsystem=windows src/*.c -I"include/" -L"lib" -lwebui-2-static -lws2_32 -Wall -Werror -luser32 -static -o webuiWin
```

**Reply by AlbertShown** [reply]
I guess you are mixing libs ABI, so make sure you are putting the Windows GCC library version in `lib` folder.
On the other hand, you can easily re-compile it from source:

```
git clone https://github.com/webui-dev/webui.git
cd webui
mingw32-make
```

Then copy `dist\webui-2.dll` into your `lib` folder.

**Reply by KaySteinhoff** [reply]
Using migw32-make results in:
```
GNUmakefile:60: Extraneous text after `else' directive
GNUmakefile:68: *** only one `else' per conditional.  Stop.
```
Resolving all issues results in:
```
process_begin: CreateProcess((null), uname, ...) failed.
mingw32-make: *** No targets.  Stop.
```

Using make:
```
MAKE Version 5.2  Copyright (c) 1987, 1998 Inprise Corp.

** error -1 ** deleting release
```

**Reply by AlbertShown** [reply]
What version of MinGW you have? And OS.
Normally building from source should be fine.
